orca-queue-tck/src/main/kotlin/com/netflix/spinnaker/orca/q/QueueIntegrationTest.kt

```diff
@@ -424,6 +426,47 @@ class QueueIntegrationTest {
     }
   }
 
+  @Test
+  fun `child pipeline is prompty cancelled with the parent regardless of task backoff time`() {
+    val childPipeline = pipeline {
+      application = "spinnaker"
+      stage {
+        refId = "wait"
+        type = "wait"
+        context = mapOf("waitTime" to 60)
+      }
+    }
+    val parentPipeline = pipeline {
+      application = "spinnaker"
+      stage {
+        refId = "1"
+        type = "pipeline"
+        context = mapOf("executionId" to childPipeline.id)
+      }
+    }
+
+    repository.store(childPipeline)
+    repository.store(parentPipeline)
+
+    whenever(dummyTask.execute(argThat { refId == "1" })) doReturn TaskResult(CANCELED)
+    context.runParentToCompletion(parentPipeline, childPipeline, runner::start, repository)
+
+    repository.retrieve(PIPELINE, parentPipeline.id).apply {
+      assertThat(status == CANCELED)
+    }
+    repository.retrieve(PIPELINE, childPipeline.id).apply {
+      assertThat(stageByRef("wait").status == RUNNING)
+    }
+
+    context.runToCompletion(childPipeline, runner::start, repository)
+
+    repository.retrieve(PIPELINE, childPipeline.id).apply {
+      assertThat(isCanceled).isTrue()
+      assertThat(stageByRef("wait").wasShorterThan(10000L)).isTrue()
```

Comment:
This assert fails if `queue.push(RescheduleExecution(child))` is removed from `CancelStageHandler`.